app/router/session_occurrence.py

```diff
@@ -0,0 +1,114 @@
+    - Otherwise, returns False
+    """
+    if start_time is not None:
+        session_start_date, session_start_time = build_date_and_time(start_time)
+        current_date, current_time = get_current_datetime()
+        return session_start_date == current_date and session_start_time <= current_time
+    return True
+
+
+def has_session_ended(end_time: str):
+    """
+    Checks if session has ended
+    - If end time is given and if session end date is greater than or equal to current date, returns True
+    - Else, always returns True
+    """
+    if end_time is not None:
+        session_end_date, session_end_time = build_date_and_time(end_time)
+        current_date, current_time = get_current_datetime()
+        return session_end_date == current_date and session_end_time >= current_time
+    return True
+
+
+@router.get("/", response_model=SessionResponse)
+async def get_session_occurrence_data(request: Request):
+    """
+    This API returns session occurrence details corresponding to the provided session ID, if the ID exists in the database and if the session is open
+
+    Parameters:
+    session_id (str): The ID against which details need to be retrieved
+
+    Returns:
+    list: Returns session occurrence details if the session ID exists in the database and the session is open, false if the session is closed. If the ID does not exist, 404 is returned
+
+    Example:
+    > $BASE_URL/session/1234
+    returns [{session_data}]
+
+    > $BASE_URL/session/{closed_session_id}
+    returns {"is_session_open": False}
+
+    > $BASE_URL/session/{invalid_id}
+    returns {
+        "status_code": 404,
+        "detail": "Session ID does not exist!",
+        "headers": null
+    }
+    """
+
+    query_params = {}
+    for key in request.query_params.keys():
+        if key not in ["name", "session_id"]:
+            raise HTTPException(
+                status_code=400, detail="Query Parameter {} is not allowed!".format(key)
+            )
+        query_params[key] = request.query_params[key]
+
+    response = requests.get(session_occurrence_db_url, params=query_params)
+
+    if response.status_code == 200:
+        if len(response.json()) != 0:
+            session_occurrence_data = response.json()
+            matched_session_occurrences = [
+                session_occurrence
+                for session_occurrence in session_occurrence_data
+                if has_session_started(session_occurrence["start_time"])
+                and has_session_ended(session_occurrence["end_time"])
+            ]
+            if len(matched_session_occurrences) > 0:
+                response = requests.get(session_db_url, params=query_params)
+
+                if response.status_code == 200:
+                    session_data = response.json()[0]
+                    if session_data["is_active"]:
+                        session_data["is_session_open"] = True
```

Comment:
Why renaming `is_active` to `is_session_open` here?

Reply:
It's not renaming. `is_active` is an attribute of the session table, which tells if the session is active. `is_session_open` is something the backend tells the frontend if the session is open at the moment.